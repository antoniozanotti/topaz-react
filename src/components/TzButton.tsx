import React from "react";
import * as heroIcons from "@heroicons/react/20/solid";
type IconName = keyof typeof heroIcons;

export interface TzButtonProps {
  /**
   * The label of the button.
   */
  label?: string;
  /**
   * The icon of the button.
   */
  iconName?: IconName;
  /**
   * Is the icon after the label?
   */
  isIconAfterLabel?: boolean;
  /**
   * The size of the button.
   */
  size?: "small" | "medium" | "large";
  /**
   * The variant of the button.
   */
  variant?: "accent" | "primary" | "secondary" | "negative" | "black" | "white";
  /**
   * The style of the button.
   */
  style?: "fill" | "outline";
  /**
   * Is the button disabled?
   */
  isDisabled?: boolean;
  /**
   * Is the button loading?
   */
  isLoading?: boolean;
}

export const TzButton = ({
  label,
  iconName,
  isIconAfterLabel = false,
  size = "medium",
  variant = "accent",
  style = "fill",
  isDisabled = false,
  isLoading = false
}: TzButtonProps) => {
  let variantStyleMode = variant;
  if (document.documentElement.classList.contains("dark")) {
    if (variant == "primary") {
      variantStyleMode = "secondary";
    } else if (variant == "secondary") {
      variantStyleMode = "primary";
    } else if (variant == "black") {
      variantStyleMode = "white";
    } else if (variant == "white") {
      variantStyleMode = "black";
    }
  }

  /* button classes */
  let buttonClasses =
    "rounded-full font-medium flex justify-center items-center";

  // focus
  buttonClasses += " focus:outline focus:outline-offset-2 focus:outline-2";

  // size
  if (size == "small") {
    // size -> small
    buttonClasses +=
      " text-sm min-h-[theme(spacing.11)] min-w-[theme(spacing.11)] lg:text-xs lg:min-h-[theme(spacing.9)] lg:min-w-[theme(spacing.9)]";
  } else if (size == "medium") {
    // size -> medium
    buttonClasses +=
      " text-base min-h-[theme(spacing.12)] min-w-[theme(spacing.12)] lg:text-sm lg:min-h-[theme(spacing.10)] lg:min-w-[theme(spacing.10)]";
  } else if (size == "large") {
    // size -> large
    buttonClasses +=
      " text-lg min-h-[theme(spacing.14)] min-w-[theme(spacing.14)] lg:text-base lg:min-h-[theme(spacing.11)] lg:min-w-[theme(spacing.11)]";
  }

  // padding only with text
  if (label) {
    if (size == "small") {
      buttonClasses += " px-6 lg:px-4";
    } else if (size == "medium") {
      buttonClasses += " px-7 lg:px-5";
    } else if (size == "large") {
      buttonClasses += " px-8 lg:px-6";
    }
  }

  // variants and styles
  if (style == "fill") {
    // fill
    if (variantStyleMode == "accent") {
      // fill -> accent
      buttonClasses +=
        " bg-accent hover:bg-accent-600 active:bg-accent-700 text-light dark:text-dark focus:outline-accent";
    } else if (variantStyleMode == "primary") {
      // fill -> primary
      buttonClasses +=
        " bg-primary hover:bg-primary-600 active:bg-primary-700 text-light focus:outline-primary";
    } else if (variantStyleMode == "secondary") {
      // fill -> secondary
      buttonClasses +=
        " bg-secondary hover:bg-secondary-600 active:bg-secondary-700 text-primary focus:outline-secondary";
    } else if (variantStyleMode == "negative") {
      // fill -> negative
      buttonClasses +=
        " bg-negative hover:bg-negative-600 active:bg-negative-700 text-light dark:text-dark focus:outline-negative";
    } else if (variantStyleMode == "black") {
      // fill -> black
      buttonClasses +=
        " bg-dark hover:bg-dark-700 active:bg-dark-600 text-light focus:outline-dark";
    } else if (variantStyleMode == "white") {
      // fill -> white
      buttonClasses +=
        " bg-light hover:bg-light-600 active:bg-light-700 text-dark focus:outline-light";
    }
  } else if (style == "outline") {
    buttonClasses += " outline outline-px";
    // outline
    if (variantStyleMode == "accent") {
      // outline -> accent
      buttonClasses +=
        " outline-accent hover:outline-accent-600 active:outline-accent-700 active: text-accent";
    } else if (variantStyleMode == "primary") {
      // outline -> primary
      buttonClasses +=
        " outline-primary hover:outline-primary-600 active:outline-primary-700 text-primary";
    } else if (variantStyleMode == "secondary") {
      // outline -> secondary
      buttonClasses +=
        " outline-secondary hover:outline-secondary-600 active:outline-secondary-700 text-primary";
    } else if (variantStyleMode == "negative") {
      // outline -> negative
      buttonClasses +=
        " outline-negative hover:outline-negative-600 active:outline-negative-700 text-negative";
    } else if (variantStyleMode == "black") {
      // outline -> black
      buttonClasses +=
        " outline-dark hover:outline-dark-700 active:outline-dark-600 text-dark";
    } else if (variantStyleMode == "white") {
      // outline -> white
      buttonClasses +=
        " outline-light hover:outline-light-600 active:outline-light-700 text-light";
    }
  }

  // is disabled
  if (isDisabled || isLoading) {
    buttonClasses += " opacity-50 pointer-events-none";
  }

  /* icon classes */
  let iconClasses;
  // size
  if (size == "small") {
    iconClasses = "h-6 w-6 lg:h-4 lg:w-4";
  } else if (size == "medium") {
    iconClasses = "h-7 w-7 lg:h-5 lg:w-5";
  } else if (size == "large") {
    iconClasses = "h-8 w-8 lg:h-6 lg:w-6";
  }
  // is loading
  if (isLoading) {
    iconClasses += " animate-spin";
  }

  /* label classes */
  let labelClasses;
  // padding between icon and text
  if (iconName) {
    if (!isIconAfterLabel) {
      labelClasses = "pl-1.5";
    } else {
      labelClasses = "pr-1.5";
    }
  }

  // label
  if (isLoading) {
    label = "loading";
  }

  // icon
  let iconComponentName;
  if (isLoading) {
    iconComponentName = heroIcons["ArrowPathIcon"];
  } else {
    iconComponentName = iconName ? heroIcons[iconName] : "";
  }

  return (
    <button className={buttonClasses} disabled={isDisabled}>
      {iconName &&
        !isIconAfterLabel &&
        React.createElement(iconComponentName, {
          className: iconClasses,
        })}
      {label && <span className={labelClasses}>{label}</span>}
      {iconName &&
        isIconAfterLabel &&
        React.createElement(iconComponentName, {
          className: iconClasses,
        })}
    </button>
  );
};
