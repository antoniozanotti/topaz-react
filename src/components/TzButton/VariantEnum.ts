export enum VariantEnum {
  accent = 'accent',
  primary = 'primary',
  secondary = 'secondary',
  negative = 'negative',
  dark = 'dark',
  light = 'light',
}
