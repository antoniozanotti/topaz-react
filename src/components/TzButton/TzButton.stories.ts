import type { Meta, StoryObj } from "@storybook/react";
import { TzButton } from "./TzButton";

/**
 * The TzButton component has styles for hover, active and focus events, try it.
 */
const meta: Meta<typeof TzButton> = {
  component: TzButton,
  tags: ["autodocs"],
  args: {
    label: "Button",
    iconName: undefined,
    isIconAfterLabel: false,
    variant: "accent",
    filled: true,
    isLoading: false,
    disabled: false
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const OnlyLabel: Story = {
  name: "Only Label",
  args: {
    label: "Home",
  },
};

export const OnlyIcon: Story = {
  name: "Only Icon",
  args: {
    label: "",
    iconName: "HomeIcon",
  },
};

export const LabelIconOnLeft: Story = {
  name: "Label and Icon on the left",
  args: {
    label: "Home",
    iconName: "HomeIcon",
  },
};

export const LabelIconOnRight: Story = {
  name: "Label and Icon on the right",
  args: {
    ...LabelIconOnLeft.args,
    isIconAfterLabel: true,
  },
};

export const VariantAccent: Story = {
  name: "Variant: Accent",
  args: {
    label: "Shop Now",
    iconName: "ShoppingBagIcon",
    variant: "accent",
  },
};

export const VariantPrimary: Story = {
  name: "Variant: Primary",
  args: {
    label: "Sign up",
    variant: "primary",
  },
};

export const VariantSecondary: Story = {
  name: "Variant: Secondary",
  args: {
    label: "Log in",
    variant: "secondary",
  },
};

export const VariantNegative: Story = {
  name: "Variant: Negative",
  args: {
    label: "Delete Account",
    iconName: "XMarkIcon",
    variant: "negative",
  },
};

export const VariantDark: Story = {
  name: "Variant: Dark",
  parameters: {
    backgrounds: {
      disable: false,
    },
  },
  args: {
    label: "Share",
    iconName: "ShareIcon",
    variant: "dark",
  },
};

export const VariantLight: Story = {
  name: "Variant: Light",
  parameters: {
    backgrounds: {
      disable: false,
    },
  },
  args: {
    label: "Share",
    iconName: "ShareIcon",
    variant: "light",
  },
};

export const FilledTrue: Story = {
  name: "Filled: true",
  parameters: {
    backgrounds: {
      disable: false,
    },
  },
  args: {
    iconName: "Bars4Icon",
    label: "Menu",
    variant: "primary",
  },
};

export const FilledFalse: Story = {
  name: "Filled: false",
  parameters: {
    backgrounds: {
      disable: false,
    },
  },
  args: {
    ...FilledTrue.args,
    filled: false,
  },
};

export const Disable: Story = {
  name: "Disable",
  args: {
    iconName: "LockClosedIcon",
    label: "Locked",
    disabled: true,
  },
};

export const Loading: Story = {
  name: "Loading",
  args: {
    iconName: "ArrowPathIcon",
    label: "loading",
    isLoading: true,
  },
};

export const SmallMobile: Story = {
  name: "Small Mobile",
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  },
  args: {
    label: "Home",
    iconName: "HomeIcon",
  },
};

export const LargeMobile: Story = {
  name: "Large Mobile",
  parameters: {
    viewport: {
      defaultViewport: "mobile2",
    },
  },
  args: {
    label: "Home",
    iconName: "HomeIcon",
  },
};

export const Tablet: Story = {
  name: "Tablet",
  parameters: {
    viewport: {
      defaultViewport: "tablet",
    },
  },
  args: {
    label: "Home",
    iconName: "HomeIcon",
  },
};

export const Desktop: Story = {
  name: "Desktop",
  args: {
    label: "Home",
    iconName: "HomeIcon",
  },
};
