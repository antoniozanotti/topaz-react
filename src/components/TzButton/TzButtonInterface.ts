import * as heroIcons from '@heroicons/react/20/solid';
import { VariantEnum } from './VariantEnum';

export type TzButtonInterface = React.ComponentProps<'button'> & {
  label?: string;
  iconName?: keyof typeof heroIcons;
  isIconAfterLabel?: boolean;
  variant?: keyof typeof VariantEnum;
  filled?: boolean;
  isLoading?: boolean;
  disabled?: boolean;
}