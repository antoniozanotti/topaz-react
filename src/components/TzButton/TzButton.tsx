import React from 'react';
import { TzIcon } from '../TzIcon/TzIcon';
import { VariantEnum as IconVariant } from '../TzIcon/VariantEnum';
import { useVariantStyles, useFocusStyles } from '../useStyles';
import { VariantEnum } from './VariantEnum';
import { TzButtonInterface } from './TzButtonInterface';

export function TzButton({
  label,
  iconName,
  isIconAfterLabel = false,
  variant = VariantEnum.accent,
  filled = true,
  isLoading = false,
  disabled = false,
  ...props
}: TzButtonInterface) {
  const baseStyles = 'rounded-full font-medium';
  const alignStyles = 'flex gap-x-2 justify-center place-items-center';
  const focusStyles = useFocusStyles();
  const variantStyles = useVariantStyles(variant, filled);
  const sizeStyles = useSizeStyles(label);
  const disabledStyles = useDisabledStyles(isLoading, disabled);
  const buttonStyles = `${baseStyles} ${alignStyles} ${focusStyles} ${variantStyles} ${sizeStyles} ${disabledStyles}`;
  props.className = props.className
    ? `${buttonStyles} ${props.className}`
    : buttonStyles;

  const iconVariant = useIconVariant(variant, filled);
  let iconBaseStyles = isIconAfterLabel ? 'order-last' : '';
  if (isLoading) {
    iconBaseStyles += ' animate-spin';
    label = 'loading';
    iconName = 'ArrowPathIcon';
  }

  return (
    <button {...props} disabled={disabled || isLoading}>
      {iconName &&
        React.createElement(TzIcon, {
          iconName: iconName,
          variant: iconVariant,
          className: iconBaseStyles,
        })}
      {label && <span>{label}</span>}
    </button>
  );
}

function useSizeStyles(label?: string) {
  let buttonSizeStyles = 'h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px]';
  if (label) {
    buttonSizeStyles += ' px-[27px] lg:px-[36px]';
  }
  return buttonSizeStyles;
}

function useDisabledStyles(isLoading?: boolean, disabled?: boolean) {
  return isLoading || disabled ? 'opacity-50 pointer-events-none' : '';
}

function useIconVariant(
  variant?: keyof typeof VariantEnum,
  filled?: boolean
): keyof typeof IconVariant | undefined {
  if (!filled) {
    return variant;
  }
  switch (variant) {
    case VariantEnum.accent:
      return IconVariant['inverse-accent'];
    case VariantEnum.primary:
      return IconVariant['inverse-primary'];
    case VariantEnum.secondary:
      return IconVariant['inverse-secondary'];
    case VariantEnum.negative:
      return IconVariant['inverse-negative'];
    case VariantEnum.dark:
      return IconVariant['inverse-dark'];
    case VariantEnum.light:
      return IconVariant['inverse-light'];
  }
  return undefined;
}
