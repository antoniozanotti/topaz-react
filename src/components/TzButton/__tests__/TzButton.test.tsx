import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TzButton } from '../TzButton';

const basicTestButton = () => {
  const button = screen.getByRole('button');
  expect(button).toBeInTheDocument();
  return button;
};

const basicTestLabel = (button: HTMLElement, text: string) => {
  expect(button).toHaveTextContent(text);
};

const basicTestIcon = (button: HTMLElement) => {
  const icon = button.children[0];
  expect(icon).toBeInstanceOf(SVGElement);
  return icon;
};

describe('TzButton Component', () => {
  it('should render only label', () => {
    const { container } = render(<TzButton label="Home" />);

    const button = basicTestButton();

    basicTestLabel(button, 'Home');
    expect(container).toMatchSnapshot();
  });

  it('should render only icon', () => {
    const { container } = render(<TzButton label="" iconName="HomeIcon" />);

    const button = basicTestButton();

    basicTestIcon(button);
    expect(container).toMatchSnapshot();
  });

  it('should render label and icon on the left', () => {
    const { container } = render(<TzButton label="Home" iconName="HomeIcon" />);

    const button = basicTestButton();

    basicTestLabel(button, 'Home');
    basicTestIcon(button);
    expect(container).toMatchSnapshot();
  });

  it('should render label and icon on the right', () => {
    const { container } = render(<TzButton label="Home" iconName="HomeIcon" isIconAfterLabel />);

    const button = basicTestButton();

    basicTestLabel(button, 'Home');

    const icon = basicTestIcon(button);

    expect(icon).toHaveClass('order-last');
    expect(container).toMatchSnapshot();
  });

  it('should render disable state', () => {
    const { container } = render(
      <TzButton label="Locked" iconName="LockClosedIcon" disabled={true} />
    );

    const button = basicTestButton();

    expect(button).toBeDisabled();
    basicTestLabel(button, 'Locked');
    basicTestIcon(button);
    expect(container).toMatchSnapshot();
  });

  it('should render loading state', () => {
    const { container } = render(<TzButton isLoading />);

    const button = basicTestButton();
    expect(button).toBeDisabled();

    basicTestLabel(button, 'loading');

    const icon = basicTestIcon(button);
    expect(icon).toHaveClass('animate-spin');
    expect(container).toMatchSnapshot();
  });
});
