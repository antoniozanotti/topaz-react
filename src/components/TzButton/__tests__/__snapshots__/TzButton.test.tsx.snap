// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TzButton Component should render disable state 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] opacity-50 pointer-events-none"
    disabled=""
  >
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent "
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        LockClosedIcon
      </title>
      <path
        d="M16.5 10.5V6.75a4.5 4.5 0 1 0-9 0v3.75m-.75 11.25h10.5a2.25 2.25 0 0 0 2.25-2.25v-6.75a2.25 2.25 0 0 0-2.25-2.25H6.75a2.25 2.25 0 0 0-2.25 2.25v6.75a2.25 2.25 0 0 0 2.25 2.25Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <span>
      Locked
    </span>
  </button>
</div>
`;

exports[`TzButton Component should render label and icon on the left 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] "
  >
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent "
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        HomeIcon
      </title>
      <path
        d="m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <span>
      Home
    </span>
  </button>
</div>
`;

exports[`TzButton Component should render label and icon on the right 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] "
  >
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent order-last"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        HomeIcon
      </title>
      <path
        d="m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <span>
      Home
    </span>
  </button>
</div>
`;

exports[`TzButton Component should render loading state 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] opacity-50 pointer-events-none"
    disabled=""
  >
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent  animate-spin"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        ArrowPathIcon
      </title>
      <path
        d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0 3.181 3.183a8.25 8.25 0 0 0 13.803-3.7M4.031 9.865a8.25 8.25 0 0 1 13.803-3.7l3.181 3.182m0-4.991v4.99"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <span>
      loading
    </span>
  </button>
</div>
`;

exports[`TzButton Component should render only icon 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] "
  >
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent "
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        HomeIcon
      </title>
      <path
        d="m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;

exports[`TzButton Component should render only label 1`] = `
<div>
  <button
    class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] "
  >
    <span>
      Home
    </span>
  </button>
</div>
`;
